# Hospital In-Patient Diagnosis

 In this project, we conduct an Exploratory Data Analysis (EDA) on hospital in-patients dataset focusing on the diseases they are diagnosed with initially from the year 2018 to 2022. The data is divided into different age groups spanning from 0 to 90 years, consisting the information about the average length of stay at the hospital. The research questions to be investigated with the hospital patients dataset are as follows:

#### 1. How does the prevalence of diseases vary between male and female? 

#### 2. What are the most common diseases diagnosed among patients in different age groups? Is there any significant difference in the average length of stay at the hospital across different age groups?

#### 3. Are there any notable trends or changes in disease prevalence over the five-year period from 2018 to 2022?


First, we will start by importing all the relevant libraries that we going to use in our EDA project. 

In [27]:
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt 
import seaborn as sns
import plotly.express as px
import warnings
warnings.filterwarnings("ignore")
import plotly.graph_objects as go
from plotly.subplots import make_subplots

To import the details of the data pre-processing, we bring in our preprocessing file into this notebook.

In [28]:
from preprocessing import preprocessing_steps

In [29]:
filename = "hospital_dataset.csv"

df1 = preprocessing_steps(filename)

This is how our dataset looks like. In total, we have 2160 entries overall ranging from the year 2018 to 2022.

In [30]:
df1

,Diagnosis,Residence,Year,Gender,1 to 18,18 to 30,30 to 50,50 to 70,70 to 90
0,Infections,Baden-Wuerttemberg,2018,Male,3.2,5.6,6.9,8.6,9.4
1,Malignant Neoplasms,Baden-Wuerttemberg,2018,Male,6.5,7.6,7.7,8.2,8.2
2,Blood,Baden-Wuerttemberg,2018,Male,3.3,5.4,5.2,6.3,6.3
3,Metabolic,Baden-Wuerttemberg,2018,Male,6.0,6.4,6.4,8.4,8.6
4,Nervous System,Baden-Wuerttemberg,2018,Male,5.1,5.6,5.4,7.0,9.0
...,...,...,...,...,...,...,...,...,...
2155,Nervous System,Thueringen,2022,Both,3.5,4.2,5.1,7.2,9.7
2156,Circulatory,Thueringen,2022,Both,2.4,4.4,5.2,6.5,8.1
2157,Respiratory,Thueringen,2022,Both,4.2,5.2,6.2,8.0,9.0
2158,Disgestive,Thueringen,2022,Both,3.5,3.6,4.3,5.4,6.5


Extracting the dataset in different variables based on the gender.

In [11]:
df_male = df1[df1['Gender'] == 'Male']
df_female = df1[df1['Gender'] == 'Female']
df_both = df1[df1['Gender'] == 'Both']

In [31]:
fig_1_to_18 = px.histogram(df_male, x='Diagnosis', y='1 to 18',
                            title="Diagnosis : Distribution of Diseases by Age Group (MALE) : Year 2018 to 2022 ",
                            height=500, color_discrete_sequence=['blue'])

fig_18_to_30 = px.histogram(df_male, x='Diagnosis', y='18 to 30', color_discrete_sequence=['green'])

fig_30_to_50 = px.histogram(df_male, x='Diagnosis', y='30 to 50', color_discrete_sequence=['red'])

fig_50_to_70 = px.histogram(df_male, x='Diagnosis', y='50 to 70', color_discrete_sequence=['yellow'])

fig_70_to_90 = px.histogram(df_male, x='Diagnosis', y='70 to 90', color_discrete_sequence=['orange'])

fig_1_to_18.update_traces(opacity=0.5)  
for data in fig_18_to_30.data:
    fig_1_to_18.add_trace(data)

fig_1_to_18.update_traces(opacity=0.5)  
for data in fig_30_to_50.data:
    fig_1_to_18.add_trace(data)


fig_1_to_18.update_traces(opacity=0.5)  
for data in fig_50_to_70.data:
    fig_1_to_18.add_trace(data)

fig_1_to_18.update_traces(opacity=0.5)  
for data in fig_70_to_90.data:
    fig_1_to_18.add_trace(data)


fig_1_to_18.update_xaxes(title_text='Diagnosis')
fig_1_to_18.update_yaxes(title_text='Length of Stay')
fig_1_to_18.show()

The above histogram shows the total amount of average length of stay for Males with different diseases based on thier age groups. We can interpret that, in Males, Malignant Neoplasms occurs the most in age groups between 1 to 18, 18 to 30 and 30 to 50, with a total of 521.3 days, 621.1 days and 612.55 days of stay in the hospital respectively. Overall, according to out data, the reason of stay inside the hopisal for males is because of Malignant Neoplasms. Males stay the least amount of days in the hospital when they have a disgestive disease. 

In [39]:
df_male_blood = df_male[df_male['Diagnosis'] == "Blood"]

In [40]:
df_male_blood.describe()

,Year,1 to 18,18 to 30,30 to 50,50 to 70,70 to 90
count,80.000000,80.000000,80.000000,80.000000,80.000000,80.000000
mean,2020.000000,5.941250,6.206250,5.471250,6.098750,6.421250
std,1.423136,2.948876,1.590744,0.633894,0.397936,0.411448
min,2018.000000,1.200000,4.000000,4.000000,5.300000,5.500000
25%,2019.000000,4.000000,5.375000,5.100000,5.800000,6.200000
50%,2020.000000,5.000000,5.900000,5.400000,6.100000,6.300000
75%,2021.000000,7.050000,6.700000,5.725000,6.400000,6.600000
max,2022.000000,15.600000,15.400000,8.600000,7.100000,7.700000


## Mean values of length of stay for Males and Females

In [41]:
mean_values_male = df_male[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].mean()
mean_values_female = df_female[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].mean()
mean_df_male = pd.DataFrame({'Age Group': mean_values_male.index, 'Mean Value': mean_values_male.values, 'Gender': 'Male'})
mean_df_female = pd.DataFrame({'Age Group': mean_values_female.index, 'Mean Value': mean_values_female.values, 'Gender': 'Female'})
mean_df_combined = pd.concat([mean_df_male, mean_df_female], ignore_index=True)

fig = px.bar(mean_df_combined, x='Age Group', y='Mean Value', color='Gender', barmode='group',
             title='Mean Values of Age Groups for Males and Females',
             labels={'Mean Value': 'Mean Value', 'Age Group': 'Age Group', 'Gender': 'Gender'})

fig.update_layout(width=600,height=600)
fig.show()


The above bar plot shows the average length of stays of males and females according to their age groups. Here too, it can be seen that with an increase in the age, the length of the stay also increases. However, Females have a lower length of stay at the hospitals that that of Males for each age group. 

The avarage length of stay of males in the age group 1 to 18 is 4.53 days. For the age group 70 to 90, the value equals 8.46 days which is almost double. 
Simlarly from Females, as age increases, the nummber of days of stays increases. 

## Male - Year 2018 to 2022

In [13]:
df_male_melted = pd.melt(df_male, id_vars=['Diagnosis'], value_vars=[ '1 to 18', '18 to 30' ,'30 to 50','50 to 70','70 to 90'], 
                     var_name='Age Group', value_name='Count')


fig = px.histogram(df_male_melted, x='Diagnosis', y='Count', color='Age Group', 
                   title="Length of Stay per Disease (MALE) : Year 2018 to 2022", height=500, barmode='group')

fig.update_xaxes(title_text='Diagnosis')
fig.update_yaxes(title_text='Length of Stay')

fig.show()

Considering all the age groups, Males in the age group 70 to 90 stay the longest inside the Hospitals. In fact, for all diseases except of diseases caused by Blood and Malignant Neoplasms, show an increase in the length of stays when age increases.

For Blood, the sum of avarage length of stay for different age groups is pretty much the same, ranging from 437.72 to 513.375 days. For age 1 to 18, the avarage length of stay for a male of age under 18, having a blood disease is 5.94 days. 6.19 days for males in age group 18 to 30, 5.47 days in the age group 30 to 50, 6.10 in the age group 50 to 70 and 6.41 in the age group 70 to 90.

## Female - Year 2018 to 2022

In [14]:
df_female_melted = pd.melt(df_female, id_vars=['Diagnosis'], value_vars=[ '1 to 18', '18 to 30' ,'30 to 50','50 to 70','70 to 90'], 
                     var_name='Age Group', value_name='Count')

fig = px.histogram(df_female_melted, x='Diagnosis', y='Count', color='Age Group', 
                   title="Length of Stay per Disease (FEMALE) : Year 2018 to 2022", height=500, barmode='group')
fig.update_xaxes(title_text='Diagnosis')
fig.update_yaxes(title_text='Length of Stay')

fig.show()

Simlarly, Females in the age group 70 to 90 stay the longest inside the Hospitals. Lenght of stay due to Metabolic diseases in the age group 1 to 18 for females is high. Length of stay in males for Maglignant Neoplasms was pretty much the same for all age groups, but here we can see that females in the age group 50 to 70 and 70 to 90 have a longer lenght of stay at the hospitals.

## Considering only years 2018 and 2022

In [15]:
# Extracting data for the year 2018 and 2022
df2018_2022 = df1[(df1["Year"] == 2018) | (df1["Year"] == 2022)]
df2018_2022

,Diagnosis,Residence,Year,Gender,1 to 18,18 to 30,30 to 50,50 to 70,70 to 90
0,Infections,Baden-Wuerttemberg,2018,Male,3.2,5.6,6.9,8.6,9.4
1,Malignant Neoplasms,Baden-Wuerttemberg,2018,Male,6.5,7.6,7.7,8.2,8.2
2,Blood,Baden-Wuerttemberg,2018,Male,3.3,5.4,5.2,6.3,6.3
3,Metabolic,Baden-Wuerttemberg,2018,Male,6.0,6.4,6.4,8.4,8.6
4,Nervous System,Baden-Wuerttemberg,2018,Male,5.1,5.6,5.4,7.0,9.0
...,...,...,...,...,...,...,...,...,...
2155,Nervous System,Thueringen,2022,Both,3.5,4.2,5.1,7.2,9.7
2156,Circulatory,Thueringen,2022,Both,2.4,4.4,5.2,6.5,8.1
2157,Respiratory,Thueringen,2022,Both,4.2,5.2,6.2,8.0,9.0
2158,Disgestive,Thueringen,2022,Both,3.5,3.6,4.3,5.4,6.5


In [16]:
selected_diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Respiratory', 'Disgestive', 
                     'Musculoskeletal', 'Nervous System', 'Metabolic', 'Circulatory']
df_male_filtered = df_male[df_male['Diagnosis'].isin(selected_diseases)]
df_female_filtered = df_female[df_female['Diagnosis'].isin(selected_diseases)]



df_male_melted = pd.melt(df_male_filtered, id_vars=['Diagnosis'], value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                         var_name='Age Group', value_name='Count')

df_female_melted = pd.melt(df_female_filtered, id_vars=['Diagnosis'], value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                           var_name='Age Group', value_name='Count')

df_combined = pd.concat([df_male_melted.assign(Gender='Male'), df_female_melted.assign(Gender='Female')])


fig = px.histogram(df_combined, x='Diagnosis', y='Count', color='Gender', 
                   facet_col='Age Group', barmode='group',
                   title="Age Group Distribution by Diagnosis")

fig.show()


The plot above shows the total length of stay of Males and Females accross all age groups for all the 9 diseases. This gives a clear and a full view of the distribution from our dataset. Bars in the age group 70 to 90 cover almost the entire plot area, and values are reasing upt0 800 total days. Histogram for age group 18 to 30 and 30 to 50 are very similar to each other. In very rare cases do Females stay longer than Males, for example in the first and the last plot area in the age group 1 to 18 and 70 to 90, for disease Malignant Neoplasms and Muscaloskeletal.

## Pie Chart - Distribution in Percentages

In [42]:
selected_disease = 'Infections'
df_male_filtered = df_male[df_male['Diagnosis'] == selected_disease]
df_female_filtered = df_female[df_female['Diagnosis'] == selected_disease]

age_group_counts_male = df_male_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()
age_group_counts_female = df_female_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]],
                    subplot_titles=['Males', 'Females'])
fig.add_trace(go.Pie(labels=age_group_counts_male.index, values=age_group_counts_male.values, name='Male'), 
              row=1, col=1)
fig.add_trace(go.Pie(labels=age_group_counts_female.index, values=age_group_counts_female.values, name='Female'), 
              row=1, col=2)
fig.update_layout(title='Distribution by Age Group for Males and Females : Infections')
fig.show()


For diseases cased by infections, 30.4% of Females and 27.6% of Males in the age group 70 to 90 contribute to the highest length of stay when comparing to all other age groups.

In [43]:
selected_disease = 'Musculoskeletal'
df_male_filtered = df_male[df_male['Diagnosis'] == selected_disease]
df_female_filtered = df_female[df_female['Diagnosis'] == selected_disease]

age_group_counts_male = df_male_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()
age_group_counts_female = df_female_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]],
                    subplot_titles=['Males', 'Females'])
fig.add_trace(go.Pie(labels=age_group_counts_male.index, values=age_group_counts_male.values, name='Male'), 
              row=1, col=1)
fig.add_trace(go.Pie(labels=age_group_counts_female.index, values=age_group_counts_female.values, name='Female'), 
              row=1, col=2)
fig.update_layout(title='Distribution by Age Group for Males and Females : Musculoskeletal Diseases')
fig.show()

An increase in the precentages of lenght of stay for Males and Femals is seen for the age group 1 to 18 due to Musculoskeletal Diseases, when compared to diseases casued by infections. 14.2% insteaed of 9.32% in males, and 13.9% insteaed of 10.4% in females.

In [44]:
selected_disease = 'Blood'
df_male_filtered = df_male[df_male['Diagnosis'] == selected_disease]
df_female_filtered = df_female[df_female['Diagnosis'] == selected_disease]

age_group_counts_male = df_male_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()
age_group_counts_female = df_female_filtered[['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90']].sum()

fig = make_subplots(rows=1, cols=2, specs=[[{'type':'domain'}, {'type':'domain'}]],
                    subplot_titles=['Males', 'Females'])
fig.add_trace(go.Pie(labels=age_group_counts_male.index, values=age_group_counts_male.values, name='Male'), 
              row=1, col=1)
fig.add_trace(go.Pie(labels=age_group_counts_female.index, values=age_group_counts_female.values, name='Female'), 
              row=1, col=2)
fig.update_layout(title='Distribution by Age Group for Males and Females : Blood Disease')
fig.show()

The lenght of stays for males and females due to blood diseases is pretty much equally distributed for all different age groups.

In [34]:
df_2018 = df1[(df1['Year'] == 2018)]
df_2022 = df1[(df1['Year'] == 2022)]
diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Respiratory', 'Disgestive', 'Musculoskeletal', 'Nervous System', 'Metabolic', 'Circulatory']
figures = []

for disease in diseases:

    df_disease_2018 = df_2018[df_2018['Diagnosis'] == disease]
    df_disease_2022 = df_2022[df_2022['Diagnosis'] == disease]

    df_disease_melted_2018 = pd.melt(df_disease_2018, id_vars=['Diagnosis'], 
                                      value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                                      var_name='Age Group', value_name='Count')
    
    df_disease_melted_2022 = pd.melt(df_disease_2022, id_vars=['Diagnosis'], 
                                      value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                                      var_name='Age Group', value_name='Count')
    

    df_disease_combined = pd.concat([df_disease_melted_2018.assign(Year=2018), 
                                     df_disease_melted_2022.assign(Year=2022)])
    
    
    fig = px.box(df_disease_combined, x='Age Group', y='Count', color='Year', 
                 title=f"Age Group Distribution of {disease} for year 2018 and 2022")
    
   
    figures.append(fig)

for fig in figures:
    fig.show()


The above generated plots describe the distribution of diseases for the year 2018 and 2022 across different age groups. 
From the above plot of Infection disease, we can see that the length of stay rates were higher in 2018 compared to 2022, with a decrease observed in age groups 1 to 18, 18 to 30, and 70 to 90, and an increase in the 30 to 50 age group. While the interquartile range remains similar for the 50 to 70 age group, the maximum value suggests a higher prevalence in 2018.

From the Malignant Neoplasm diseases, we can see a decrease in the length of stay across all age groups from 1 to 90. However, there are outliers present in the age group 1 to 18, suggesting variability in the length of stay of malignant neoplasm within this age range. This pattern suggests a general trend of declining diagnoses of Malignant Neoplasm diseases across all age groups over the specified time period. The presence of outliers in the age group 1 to 18 indicates that while there is an overall decrease, there are instances of variability or exceptions within this specific age range.

From the Blood diseases boxplot, we can see that there is a decrease in blood disease among the age groups 30 to 50, 50 to 70, and 70 to 90, while the age group 1 to 18 shows an increase over the same period. The average length of stay for the age group 18 to 30 remains relatively stable across the years. This suggests evolving patterns in the prevalence of blood diseases across various age groups over time. The decrease in diagnoses among older age groups and the increase among younger individuals may reflect shifts in disease incidence or changes in diagnostic practices.

The boxplot for the respiratory diseases, suggests that the respiratory diseases across the age groups 1 to 18 and 18 to 30 has decreased over time, while the diseases increased for the age groups 30 to 50, 50 to 70, and 70 to 90. The presence of outliers indicates potential variability or exceptional cases within certain age groups, warranting further investigation into underlying factors contributing to these trends.

The disgestive diseases boxplot suggests there has been a significant decline in the length of stay over time where the age groups from 18 to 90 years saw a decline while it was a bit higher in the age groups 1 to 18, signifing that they are more prone to digestive diseases and need in-patient treatment as compared to the other age groups. Also, seen in the circulatory and metabolic diseases.

The musculoskeletal diseases has decrease over time as well for the age groups indicating that there may be some decline in the length of stay over the years suggesting that the treatments may have worked while the length of stay for the nervous system diseases has increased over time especially for the elderly. 

In [184]:
dfinfec = df_2018[df_2018['Diagnosis'] == "Infections"]


In [185]:
dfinfec.describe()

,Year,1 to 18,18 to 30,30 to 50,50 to 70,70 to 90
count,48.0,48.000000,48.000000,48.000000,48.000000,48.000000
mean,2018.0,3.187500,5.370833,6.329167,8.597917,9.620833
std,0.0,0.482734,1.303997,0.901525,0.753288,0.554271
min,2018.0,2.400000,3.600000,5.100000,7.300000,8.600000
25%,2018.0,2.900000,4.600000,5.675000,8.000000,9.200000
50%,2018.0,3.100000,5.100000,6.200000,8.600000,9.600000
75%,2018.0,3.300000,5.800000,6.800000,9.100000,10.000000
max,2018.0,4.600000,9.600000,9.600000,10.900000,11.200000


In [158]:
selected_year = 2018

df_selected_year = df1[df1['Year'] == selected_year]

diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Nervous System', 
            'Metabolic', 'Circulatory', 'Respiratory', 'Disgestive', 'Musculoskeletal']

df_melted = pd.melt(df_selected_year, id_vars=['Diagnosis'], 
                    value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                    var_name='Age Group', value_name='Count')

fig = px.box(df_melted, x='Diagnosis', y='Count', color='Diagnosis', 
             title=f"Box plot of Diseases for {selected_year}",
             category_orders={'Diagnosis': diseases})

fig.show()


- For year 2018

In the above box plot, Infections has the largest Inter Quartile Range (IQR) or 4.36, that is, most of the data 50% of is is present inside Quartile 1 and Quartile 3, while Malignant Neoplasms has the smallest IQR with 1.49. Blood has an IQR of 1.5.
These two diseases have outlier values. 

Infections also has the widest range of values ranging from a minimum of 2.4 days to a maximum of 11.175 days. The median is 6.55 days. 
Values for the length of avarage stay for diseases caused due to Nervous System and Respiratory issues also have a wider range of values. Respiratory has a median of 4.637.

Lenght of stay has a miximum value of 7.825, which is the lowest amongst all the diseases.

In [46]:
selected_year = 2022

df_selected_year = df_male[df_male['Year'] == selected_year]

diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Nervous System', 
            'Metabolic', 'Circulatory', 'Respiratory', 'Disgestive', 'Musculoskeletal']

df_melted = pd.melt(df_selected_year, id_vars=['Diagnosis'], 
                    value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                    var_name='Age Group', value_name='Count')

fig = px.box(df_melted, x='Diagnosis', y='Count', color='Diagnosis', 
             title=f"Box plot for Diseases for {selected_year}",
             category_orders={'Diagnosis': diseases})

fig.show()


- For year 2022

Here too, Infections has the largest Inter Quartile Range (IQR) 5.825, that is, most of the data 50% of is is present inside Quartile 1 and Quartile 3, while Malignant Neoplasms has the smallest IQR with 0.925. Blood has an IQR of 1.285. Only these two diseases w.r.t length of stay have outlier values. Respiratory has the second largest IQR. 

Here length of stay due to Nervous System issues has the widest range of values, with a maximum value of 12.05 and a minimum of 3.1. 

Digestive has the lowest median value of 4.1375

In [47]:
selected_year = 2018

df_male_selected_year = df_male[df_male['Year'] == selected_year]
df_female_selected_year = df_female[df_female['Year'] == selected_year]

diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Nervous System', 
            'Metabolic', 'Circulatory', 'Respiratory', 'Disgestive', 'Musculoskeletal']

df_male_melted = pd.melt(df_male_selected_year, id_vars=['Diagnosis'], 
                          value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                          var_name='Age Group', value_name='Count')
df_male_melted['Gender'] = 'Male'

df_female_melted = pd.melt(df_female_selected_year, id_vars=['Diagnosis'], 
                            value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                            var_name='Age Group', value_name='Count')
df_female_melted['Gender'] = 'Female'

df_combined = pd.concat([df_male_melted, df_female_melted])

fig = px.box(df_combined, x='Diagnosis', y='Count', color='Gender',
             title=f"Age Group Distribution of Diseases for {selected_year} (Male vs Female)",
             category_orders={'Diagnosis': diseases})

fig.show()


In [48]:
selected_year = 2022

df_male_selected_year = df_male[df_male['Year'] == selected_year]
df_female_selected_year = df_female[df_female['Year'] == selected_year]

diseases = ['Infections', 'Malignant Neoplasms', 'Blood', 'Nervous System', 
            'Metabolic', 'Circulatory', 'Respiratory', 'Disgestive', 'Musculoskeletal']

df_male_melted = pd.melt(df_male_selected_year, id_vars=['Diagnosis'], 
                          value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                          var_name='Age Group', value_name='Count')
df_male_melted['Gender'] = 'Male'

df_female_melted = pd.melt(df_female_selected_year, id_vars=['Diagnosis'], 
                            value_vars=['1 to 18', '18 to 30', '30 to 50', '50 to 70', '70 to 90'], 
                            var_name='Age Group', value_name='Count')
df_female_melted['Gender'] = 'Female'

df_combined = pd.concat([df_male_melted, df_female_melted])

fig = px.box(df_combined, x='Diagnosis', y='Count', color='Gender',
             title=f"Age Group Distribution of Diseases for {selected_year} (Male vs Female)",
             category_orders={'Diagnosis': diseases})

fig.show()
